ibis/backends/sql/__init__.py

```diff
@@ -427,17 +427,19 @@ def _build_insert_from_table(
         # ordering.
         source_cols = source.columns
         columns = (
-            source_cols
-            if not set(target_cols := self.get_schema(target).names).difference(
-                source_cols
-            )
-            else target_cols
+            target_cols
+            if set(target_cols := self.get_schema(target).names).difference(source_cols)
+            else source_cols
         )
 
         query = sge.insert(
             expression=self.compile(source),
             into=sg.table(target, db=db, catalog=catalog, quoted=quoted),
-            columns=[sg.to_identifier(col, quoted=quoted) for col in columns],
+            columns=[
+                sg.to_identifier(col, quoted=quoted)
+                for col in columns
+                if col in source_cols
```

Comment:
Let's make a variable for `source.schema` and then use `col in variable` to do this lookup.

Reply:
I believe the change I have made will reduce the time complexity to avoid needing to check again against the source_cols list. I have rewritten things a bit here, using a variable named `columns` to refer to the column list to be used for the insert SQL expression.